apiserver/src/api/node.rs

```diff
@@ -191,4 +202,64 @@ mod tests {
             panic!("Response did not return a body.");
         }
     }
+
+    #[tokio::test]
+    async fn test_convert_crd() {
+        let conversion_req = ConversionRequest {
+            kind: "ConversionReview".to_string(),
+            api_version: "apiextensions.k8s.io/v1".to_string(),
+            request: Request {
+                uid: "5a6adc7e-c74b-43c0-9718-293de1b104cb".to_string(),
+                desired_api_version: "brupop.bottlerocket.aws/v2".to_string(),
+                objects: vec![json!({
+                    "apiVersion": "brupop.bottlerocket.aws/v1",
+                    "kind": "BottlerocketShadow",
+                    "metadata": {
+                        "name": "brs-ip-192-168-22-145.us-west-2.compute.internal",
+                        "namespace": "brupop-bottlerocket-aws",
+                        "uid": "3153df27-6619-4b6b-bc75-adbf92ef7266",
+                        "ownerReferences": [
+                            {
+                                "apiVersion": "v1",
+                                "kind": "Node",
+                                "name": "ip-192-168-22-145.us-west-2.compute.internal",
+                                "uid": "6b714046-3b20-4a79-aaa9-27cf626a2c12"
+                            }
+                        ]
+                    },
+                    "spec": {
+                        "state": "Idle",
+                    },
+                    "status": {
+                        "current_state": "Idle",
+                        "target_version": "1.8.0",
+                        "current_version": "1.8.0"
+                    }
+
+                })],
+            },
+        };
+        let req = test::TestRequest::put()
+            .uri(CRD_CONVERT_ENDPOINT)
+            .set_json(&conversion_req)
+            .to_request();
+
+        let app = test::init_service(App::new().route(
+            CRD_CONVERT_ENDPOINT,
+            web::put().to(convert_bottlerocket_shadow_resource),
+        ))
+        .await;
+
+        let resp = test::call_service(&app, req).await;
+
+        assert!(resp.status().is_success());
+        if let AnyBody::Bytes(b) = resp.into_body() {
+            // Only check the response body can be converted to ConversionResponse.
+            // Contents of the ConversionResponse should be tested in convert_resource method.
+            let _return_response: ConversionResponse =
+                serde_json::from_slice(&b).expect("Could not parse JSON response.");
```

Comment:
If you are not going to use the object, I think you can use `::from_slice<Type>()`